src/k8s/pkg/k8sd/api/impl/k8sd.go

```diff
@@ -46,22 +45,15 @@ func GetLocalNodeStatus(ctx context.Context, s *state.State, snap snap.Snap) (ap
 	if err != nil {
 		return apiv1.NodeStatus{}, fmt.Errorf("failed to check if node is a worker: %w", err)
 	}
+
 	if isWorker {
 		clusterRole = apiv1.ClusterRoleWorker
-	} else {
-		node, err := nodeutil.GetControlPlaneNode(ctx, s, s.Name())
-		if err != nil {
-			// The node is likely in a joining or leaving phase where the role is not yet settled.
-			// Use the unknown role but still log this incident for debugging.
-			log.Printf("Failed to check if node is control-plane. This is expected if the node is in a joining/leaving phase. %v", err)
-			clusterRole = apiv1.ClusterRoleUnknown
-		} else {
-			if node != nil {
-				return *node, nil
-			}
-		}
-
+	} else if node, err := nodeutil.GetControlPlaneNode(ctx, s, s.Name()); err != nil {
+		clusterRole = apiv1.ClusterRoleUnknown
```

Comment:
Is the log here no longer necessary?

Reply:
yes, removed log lines that did not provide any useful context here and in some other places

Reply:
I would keep around the comment on when this might happen. Let's remove the logs though.